packages/core/src/cache/classes/ContourSet.ts

```diff
@@ -0,0 +1,152 @@
+import { Point3, IContourSet, IContour, ContourData } from '../../types';
+import Contour from './Contour';
+
+/**
+ * This class represents a set of contours.
+ * Usually contours are grouped together in a contour set to represent a meaningful shape.
+ */
+export class ContourSet implements IContourSet {
+  readonly id: string;
+  readonly sizeInBytes: number;
+  private color: Point3 = [200, 0, 0]; // default color
+  contours: IContour[];
+
+  constructor(props: { id: string; data: ContourData[]; color?: Point3 }) {
+    this.id = props.id;
+    this.contours = [];
+    this.color = props.color ?? this.color;
+
+    this._createEachContour(props.data);
+    this.sizeInBytes = this._getSizeInBytes();
+  }
+
+  _createEachContour(contourDataArray: ContourData[]): void {
+    for (let i = 0; i < contourDataArray.length; i++) {
+      const contour = new Contour({
+        id: `${this.id}-${i}`,
+        data: {
+          points: contourDataArray[i].points,
+          type: contourDataArray[i].type,
+        },
+        color: contourDataArray[i].color ?? this.color,
+      });
+
+      this.contours.push(contour);
+    }
+  }
+
+  _getSizeInBytes(): number {
+    let sizeInBytes = 0;
+    this.contours.forEach((contour) => {
+      sizeInBytes += contour.sizeInBytes;
+    });
+
+    return sizeInBytes;
+  }
+
+  public getColor(): Point3 {
+    // Currently, all contours in a contour set have the same color.
+    // This may change in the future.
+    return this.color;
+  }
+
+  /**
+   * This function returns the contours of the image
+   * @returns The contours of the image.
+   */
+  public getContours(): IContour[] {
+    return this.contours;
+  }
+
+  public getSizeInBytes(): number {
+    return this.sizeInBytes;
+  }
+
+  /**
+   * It returns an array of all the points in the glyph
+   * @returns An array of points.
+   */
+  public getFlatPointsArray(): Point3[] {
+    const flatPoints = [];
+    this.contours.forEach((contour) => {
+      flatPoints.push(...contour.getPoints());
```

Comment:
I am a bit suprised that `contour.getFlatPoints()` is not being used here instead.